.github/workflows/build.yml

```
@@ -0,0 +1,67 @@
+name: build and test
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+    - '**'
+    - '!**.yml'
+    - '!docs/**'
+    - '!**.md'
+    - '**/build.yml'
+  pull_request:
+    branches: [ master ]
+    paths:
+    - '**'
+    - '!**.yml'
+    - '!docs/**'
+    - '!**.md'
+    - '**/build.yml'
+
+jobs:
+  meson:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04]
```

```suggestion
        os: [ubuntu-22.04, ubuntu-24.04]
```

Would you like to see if the if the results are any different

---

I also possibly fixed in in master, so those commit should be ported over